### PR TITLE
[DOC] Tweaks for Array#reverse_each

### PR DESCRIPTION
**Pull request by BurdetteLamar**

### CODE REVIEW

File: array.c [hunk 1]
```diff
@@ -2663,47 +2663,31 @@ rb_ary_each_index(VALUE ary)
...
- *    a = [:foo, 'bar', 2]
- *    a.reverse_each {|element|  puts "#{element.class} #{element}" }
- *
- *  Output:
- *
- *    Integer 2
- *    String bar
- *    Symbol foo
+ *    symbols = []
+ *    ['a', 'b', 'c'].reverse_each {|element| symbols.push(element.to_sym) }
```
Comment (peterzhu2118) [inline array.c:2674]: What's the purpose of calling `to_sym` here?

Comment (BurdetteLamar) [reply array.c]: Example recast.

File: array.c [hunk 2]
```diff
@@ -2663,47 +2663,31 @@ rb_ary_each_index(VALUE ary)
...
- *    a = [:foo, 'bar', 2]
- *    e = a.reverse_each
- *    e # => #<Enumerator: [:foo, "bar", 2]:reverse_each>
- *    a1 = e.each {|element|  puts "#{element.class} #{element}" }
- *
- *  Output:
+ *    symbols = []
+ *    a = ['a', 'b', 'c']
+ *    a.reverse_each do |element|
+ *      symbols.push(element.to_sym)
```
Comment (peterzhu2118) [inline array.c:2682]: Ditto on the `to_sym` here.

File: array.c [hunk 3]
```diff
@@ -2663,47 +2663,29 @@ rb_ary_each_index(VALUE ary)
...
- *    e # => #<Enumerator: [:foo, "bar", 2]:reverse_each>
- *    a1 = e.each {|element|  puts "#{element.class} #{element}" }
- *
- *  Output:
+ *    a = []
+ *    ['a', 'b', 'c'].reverse_each do |element|
+ *      a.push(element)
+ *      a.clear if element.start_with?('b')
+ *    end
+ *    a # => ["a"]
```
Comment (peterzhu2118) [inline array.c:2684]: This doesn't demonstrate the array that is being iterated on is modified. It demonstrates that `a` (which is an unrelated array) is modified.

Comment (BurdetteLamar) [reply array.c]: Revised.

**Review by peterzhu2118 — APPROVED** [review]

**Pull request merged**